cores/arduino/CDCSerialClass.cpp

```diff
@@ -36,6 +36,33 @@ extern void CDCSerial_Handler(void);
 extern void serialEventRun1(void) __attribute__((weak));
 extern void serialEvent1(void) __attribute__((weak));
 
+static void cdc_mbox_isr(CurieMailboxMsg msg)
+{
+    int new_head;
+    for(int i = 0; i < 4; i++)
+    {
+        uint32_t dword = msg.data[i];
+        for(int j = 0; j <4; j++)
+        {
+            if(((uint8_t)dword) != '\0')
+            {
+                new_head = (Serial._rx_buffer->head +1) % CDCACM_BUFFER_SIZE;
```

Comment:
Surely there's some kind of 'ringbuffer_add' method? we should not be re-implementing this logic for each write to the buffer.